commit 1fd32e9c2f15eabdc7e531b1fa76f62fc5a8ca75
Author: Nikita Popov <nikita.ppv@gmail.com>
Date:   Mon Mar 11 15:35:02 2019 +0100

    Fixed bug #76717
    
    Print INT_MIN as -INT_MAX-1 to avoid it getting parsed as a float
    literal due to integer overflow.

diff --git a/ext/standard/tests/general_functions/bug76717.phpt b/ext/standard/tests/general_functions/bug76717.phpt
new file mode 100644
index 0000000000..b4bbd04a51
--- /dev/null
+++ b/ext/standard/tests/general_functions/bug76717.phpt
@@ -0,0 +1,14 @@
+--TEST--
+Bug #76717: var_export() does not create a parsable value for PHP_INT_MIN
+--FILE--
+<?php
+
+$min = eval('return '.var_export(PHP_INT_MIN, true).';');
+$max = eval('return '.var_export(PHP_INT_MAX, true).';');
+var_dump($min === PHP_INT_MIN);
+var_dump($max === PHP_INT_MAX);
+
+?>
+--EXPECT--
+bool(true)
+bool(true)
diff --git a/ext/standard/var.c b/ext/standard/var.c
index a5a88b634d..0301605610 100644
--- a/ext/standard/var.c
+++ b/ext/standard/var.c
@@ -459,6 +459,13 @@ again:
 			smart_str_appendl(buf, "NULL", 4);
 			break;
 		case IS_LONG:
+			/* INT_MIN as a literal will be parsed as a float. Emit something like
+			 * -9223372036854775807-1 to avoid this. */
+			if (Z_LVAL_P(struc) == ZEND_LONG_MIN) {
+				smart_str_append_long(buf, ZEND_LONG_MIN+1);
+				smart_str_appends(buf, "-1");
+				break;
+			}
 			smart_str_append_long(buf, Z_LVAL_P(struc));
 			break;
 		case IS_DOUBLE:
